modules/hdf5_handler/HDF5Array.h

```diff
@@ -127,6 +122,14 @@ class HDF5Array:public libdap::Array {
     void m_array_of_region_reference_hyperslab_selection(hid_t space_id, int ndim, const std::string &varname,
                                                              std::vector<std::string> &v_str,int64_t i) const;
     friend class HDF5Structure;
+
+  protected:
+    
+    // Parse constraint expression and make HDF5 coordinate point location.
+    // return number of elements to read. 
+   int64_t format_constraint(int64_t *cor, int64_t *step, int64_t *edg);
```

Comment:
If this is going to be overloaded by a child class, then it should be virtual.

Reply:
This method is not going to be overloaded.  It is declared as private in the base class. So the child class cannot use it. So I use "protected." I don't want to declare this method as a public method.  Had a conversation with James and James agrees with what I did.